selectrum.el

```diff
@@ -559,312 +544,263 @@ PRED defaults to `minibuffer-completion-predicate'."
-          (setq selectrum--ensure-centered-timer
-                (run-with-idle-timer
-                 0 nil #'selectrum--ensure-current-candidate-centered)))))))
-
 (defun selectrum--minibuffer-post-command-hook ()
   "Update minibuffer in response to user input."
+  ;; Stay within input area.
   (goto-char (max (point) selectrum--start-of-input-marker))
   (goto-char (min (point) selectrum--end-of-input-marker))
-  (save-excursion
-    (let ((inhibit-read-only t)
-          ;; Don't record undo information while messing with the
-          ;; minibuffer, as per
-          ;; <https://github.com/raxod502/selectrum/issues/31>.
-          (buffer-undo-list t)
-          (input (buffer-substring selectrum--start-of-input-marker
-                                   selectrum--end-of-input-marker))
-          (bound (marker-position selectrum--end-of-input-marker))
-          (keep-mark-active (not deactivate-mark)))
-      (unless (equal input selectrum--previous-input-string)
-        (when (and (not selectrum--preprocessed-candidates)
-                   minibuffer-completion-table)
-          ;; No candidates were passed, initialize them from
-          ;; `minibuffer-completion-table'.
-          (setq selectrum--preprocessed-candidates
-                (funcall selectrum-preprocess-candidates-function
-                         (selectrum--get-candidates-from-table))))
-        (setq selectrum--previous-input-string input)
-        ;; Reset the persistent input, so that it will be nil if
-        ;; there's no special attention needed.
-        (setq selectrum--visual-input nil)
-        (let ((cands (if (functionp selectrum--preprocessed-candidates)
-                         (funcall selectrum-preprocess-candidates-function
-                                  (let ((result
-                                         (funcall
-                                          selectrum--preprocessed-candidates
-                                          input)))
-                                    (if (stringp (car result))
-                                        result
-                                      (setq input (or (alist-get 'input result)
-                                                      input))
-                                      (setq selectrum--visual-input input)
-                                      (alist-get 'candidates result))))
-                       selectrum--preprocessed-candidates)))
-          (setq selectrum--total-num-candidates (length cands))
-          (setq selectrum--refined-candidates
-                (funcall selectrum-refine-candidates-function input cands)))
-        (when selectrum--move-default-candidate-p
-          (setq selectrum--refined-candidates
-                (selectrum--move-to-front-destructive
-                 selectrum--default-candidate
-                 selectrum--refined-candidates)))
+  ;; For some reason this resets and thus can't be set in setup hook.
+  (setq-local truncate-lines t)
```

Comment:
Oh wait, does this fix the multiline candidate display problem? That's clearly a much better solution. We could extend it by cutting each candidate off before the first newline.

Reply:
No unfortunately not. It is needed because without this if you start a recursive session and exit that again the prompt line gets pushed out of sight to the left for some reason. It's a weird bug similar to one I remembered occured for `icomplete` when inserting newline into the overlay. This is what `icomplete-vertical` does to work around this issue and it removed the bug in selectrum, too.

Only showing the first line for mult-line candidates is a good idea, I will try that.